src/StoreMeta.h

```diff
@@ -80,67 +64,96 @@ enum {
         uint16_t flags;
      \endcode
      */
-    STORE_META_STD,
-
-    /**
-     * Reserved for future hit-metering (RFC 2227) stuff
-     */
-    STORE_META_HITMETERING,
-
-    // TODO: document this TLV type code
-    STORE_META_VALID,
+    STORE_META_STD = 5,
 
     /**
      * Stores Vary request headers
      */
-    STORE_META_VARY_HEADERS,
+    STORE_META_VARY_HEADERS = 8,
 
     /**
      * Updated version of STORE_META_STD, with support for  >2GB objects.
      * As STORE_META_STD except that the swap_file_sz is a 64-bit integer instead of 32-bit.
      */
-    STORE_META_STD_LFS,
+    STORE_META_STD_LFS = 9,
 
     // TODO: document this TLV type code
-    STORE_META_OBJSIZE,
+    STORE_META_OBJSIZE = 10
 
-    STORE_META_STOREURL,    /**< the Store-ID url, if different to the normal URL */
-    STORE_META_VARY_ID,     /**< Unique ID linking variants */
-    STORE_META_END
+    // When adding values, update Store::SwapMetaTypeMax.
+    // When removing values, check Store::IgnoredSwapMetaType() and friends.
 };
 
-/// \ingroup SwapStoreAPI
-class StoreMeta
+namespace Store {
+
+/// swap meta type ID written to or loaded from Store
+using RawSwapMetaType = char;
+
+/// maximum value of a recognized swap meta type
+const int SwapMetaTypeMax = 12; // Use "inline constexpr ..." with C++17.
```

Comment:
Can we use RawSwapMetaType instead?
```suggestion
const RawSwapMetaType SwapMetaTypeMax = SwapMetaType::end_; // Use "inline constexpr ..." with C++17.
```

Reply:
I had to revert that change at d14bd7d, because SwapMetaTypeMax specifies the maximum enum type, which is int (for now).

Reply:
Yes, SwapMetaTypeMax is not the serialized type (i.e. RawSwapMetaType or char) maximum -- that maximum is 127 or 255. In general, we should limit RawSwapMetaType use/exposure as much as possible -- it is just a serialization detail that high-level code should not know about. And SwapMetaType may even stop being an integer!

As for the initialization value, I found an acceptable way to get rid of that hard-coded value in branch commit 9c0e553.

Reply:
@rousskov I agree with your sentiment there. In part my request is to help with that - since using the ```RawSwapMetaType``` type name instead of obtuse ```char``` will help highlight code that may need further work to avoid the type exposure.